src/Umbraco.Web/Models/ImageProcessorImageUrlGenerator.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+
+namespace Umbraco.Web.Models
+{
+    internal class ImageProcessorImageUrlGenerator : IImageUrlGenerator
+    {
+        public string GetImageUrl(ImageUrlGenerationOptions options)
+        {
+            var imageProcessorUrl = new StringBuilder(options.ImageUrl ?? string.Empty);
+
+            if (options.FocalPoint != null)
+            {
+                imageProcessorUrl.Append("?center=");
+                imageProcessorUrl.Append(options.FocalPoint.Top.ToString(CultureInfo.InvariantCulture));
+                imageProcessorUrl.Append(",");
+                imageProcessorUrl.Append(options.FocalPoint.Left.ToString(CultureInfo.InvariantCulture));
+                imageProcessorUrl.Append("&mode=crop");
+            }
+            else if (options.Crop != null)
+            {
+                imageProcessorUrl.Append("?crop=");
+                imageProcessorUrl.Append(options.Crop.X1.ToString(CultureInfo.InvariantCulture)).Append(",");
+                imageProcessorUrl.Append(options.Crop.Y1.ToString(CultureInfo.InvariantCulture)).Append(",");
+                imageProcessorUrl.Append(options.Crop.X2.ToString(CultureInfo.InvariantCulture)).Append(",");
+                imageProcessorUrl.Append(options.Crop.Y2.ToString(CultureInfo.InvariantCulture));
+                imageProcessorUrl.Append("&cropmode=percentage");
+            }
+            else if (options.DefaultCrop)
+            {
+                imageProcessorUrl.Append("?anchor=center");
+                imageProcessorUrl.Append("&mode=crop");
+            }
+            else
+            {
+                imageProcessorUrl.Append("?mode=" + options.ImageCropMode.ToString().ToLower());
```

Comment:
Think you should probably continue to use the `StringBuilder` here (and in similar constructs), i.e. `imageProcessorUrl.Append("?mode=").Append(options.ImageCropMode.ToString().ToLower());`